### DISCUSSION TITLE
Inputting location co-ords

### DISCUSSION THREAD

**Discussion by PhilRosa-Leeke** [meshtastic/Discussions#41] [General]
Is there a way that we can input (via typing) our location co-ords?
I know we can "use our phone", but isn't there a way we could do it without depending on our phone?

**Comment by ikearat** [comment]
Other than having a device that hosts it's own keyboard there are only two ways to pass instructions to the device. Phone app and web client. Both of these ways allow you to manually enter coordinates with some "gotchas". The firmware design as I understand it is structured to collect an initial position from a phone app or a hardwired GPS module even if you are not going to expose the position to the mesh. After the device has a "fix" you can manage the coordinates manually. There are work a rounds and and other nonsense involving manipulating channels but without a phone, web browser or physical keyboard you can't enter coordinates manually.
My understanding of this is anecdotal and through banging my head on the desk, almost just tossing the whole mess into my 'tastic can. So perhaps this reply if helpful you can use and if my post is a hay bale with a dead cow in it, an actual knowledgeable individual will chime in and educate both of us.

**Comment by ianmcorvidae** [comment]
For the fixed position setting, designed for nodes without a physical GPS or a regularly attached client (so things like solar repeater nodes on mountaintops), there's a few ways to set the position. With iOS you can only set it to the phone's current location, but on Android once the setting is selected, while it will automatically fill to the phone's position, it can be edited. With the CLI, fixed positions can be set with the `--setlat` `--setlon` and `--setalt` options (provide all three, or lat and lon without alt, in the same command). The web client doesn't support setting a fixed position at this time.

With fixed position not on, you must either have a physical GPS or a connected client (a phone, generally, though it's theoretically possible to use something custom with the client libraries).

**Reply by ikearat** [reply]
Thank you. I overlooked the CLI (command line interface) in my reply. This kind of support is why I invited a educator to the discussion :)
